plugins/obs-qsv11/obs-qsv11.c

```diff
@@ -826,8 +826,23 @@ static void obs_qsv_video_info(void *data, struct video_scale_info *info)
 	cap_resolution(obsqsv->encoder, info);
 }
 
+static mfxI64 ts_obs_to_mfx(int64_t ts, const struct video_output_info *voi)
+{
+	return ts * 90000 / voi->fps_num;
+}
+
+static int64_t ts_mfx_to_obs(mfxI64 ts, const struct video_output_info *voi)
+{
+	if (ts < 0)
+		return (ts * voi->fps_num / 90000 - voi->fps_den / 2) /
+		       voi->fps_den * voi->fps_den;
+	else
+		return (ts * voi->fps_num / 90000 + voi->fps_den / 2) /
+		       voi->fps_den * voi->fps_den;
```

Comment:
If there is an issue at non-fractional frame rate, I think we should have rounding for dividing by 90000 as well.
```suggestion
	int64_t den = 90000 * voi->fps_den;
	/* Round to the nearest integer multiple of den */
	if (ts < 0)
		return (ts * voi->fps_num - den / 2) / den * voi->fps_den;
	else
		return (ts * voi->fps_num + den / 2) / den * voi->fps_den;
```

Reply:
Implemented this change since I noticed DTS can be inaccurate even when the frame rate is integer but uncommon values such as 119 fps.